packages-ts/gauntlet-terra/src/commands/ledgerKey.ts

```diff
@@ -0,0 +1,118 @@
+import { Key, SimplePublicKey, SignatureV2, SignDoc, SignerInfo, ModeInfo } from '@terra-money/terra.js'
+import { SignMode } from '@terra-money/terra.proto/cosmos/tx/signing/v1beta1/signing'
+import LedgerTerraConnector, {ERROR_CODE, CommonResponse} from '@terra-money/ledger-terra-js'
+import TransportNodeHid from "@ledgerhq/hw-transport-node-hid"
+import { logger } from '@chainlink/gauntlet-core/dist/utils'
+import { signatureImport } from 'secp256k1';
+
+
+export class LedgerKey extends Key {
+    private path: Array<number>
+
+    private constructor(path: Array<number>) {
+        super()
+        this.path = path
+    }
+
+    public async initialize() {
+        const {ledgerConnector, terminateConnection} = await this.connectToLedger()
+
+        const response = await ledgerConnector.getPublicKey(this.path)
+        this.checkForErrors(response)
+
+        this.publicKey = new SimplePublicKey(Buffer.from(response.compressed_pk.data).toString('base64'))
+        await terminateConnection()
+    }
+
+    public static async create(path: string): Promise<LedgerKey> {
+        const pathArr = this.pathStringToArray(path)
+        const ledgerKey = new LedgerKey(pathArr)
+        await ledgerKey.initialize()
+
+        return ledgerKey
+    }
+
+    public async createSignature(signDoc: SignDoc): Promise<SignatureV2> {
```

Comment:
Is there any difference between this and the base `Key` `createSignature` function?

Reply:
yes, the difference is in data and data format supplied to the sign function: Ledger requires amino encoded